R/commented_code_linter.R

```diff
@@ -86,6 +86,10 @@ commented_code_linter <- function() {
     # ignore trailing ',' when testing for parsability
     extracted_code <- re_substitutes(extracted_code, rex(",", any_spaces, end), "")
     extracted_code <- re_substitutes(extracted_code, rex(start, any_spaces, ","), "")
+    # ignore a trailing pipe (|> or %>%) when testing parsability
+    extracted_code <- re_substitutes(extracted_code, rex("|>", any_spaces, end), "")
```

Comment:
We might combine these:

```r
rex(or(",", "|>", "%>%"), any_spaces, end)
```

Reply:
yep nice idea - thanks :)

Just to sense check this is the behaviour under your suggestion - first two not parsable so fine i think

``` r
lintr::lint(
  text = "# f() , |>",
  linters = lintr::commented_code_linter()
)
#> ℹ No lints found.
```

``` r

lintr::lint(
  text = "# f() |> ,",
  linters = lintr::commented_code_linter()
)
#> ℹ No lints found.
```

``` r

lintr::lint(
  text = "# f() |>",
  linters = lintr::commented_code_linter()
)
#> <text>:1:3: style: [commented_code_linter] Remove commented code.
#> # f() |>
#>   ^~~~~~
```

``` r

lintr::lint(
  text = "# f() ,",
  linters = lintr::commented_code_linter()
)
#> <text>:1:3: style: [commented_code_linter] Remove commented code.
#> # f() ,
#>   ^~~~~
```

<sup>Created on 2024-10-22 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
Yea I think it's unlikely to encounter multiple of these in the same comment & esp. that removing them will produce something parseable. I am fine with the above.